Source/Scene/Cesium3DTileStyle.js

```diff
@@ -0,0 +1,451 @@
+     * {@link https://github.com/AnalyticalGraphicsInc/3d-tiles/tree/master/Styling|3D Tiles Styling language}.
+     *
+     * @alias Cesium3DTileStyle
+     * @constructor
+     *
+     * @param {String|Object} [data] The url of a style or an object defining a style.
+     *
+     * @example
+     * tileset.style = new Cesium.Cesium3DTileStyle({
+     *     color : {
+     *         conditions : [
+     *             ['${Height} >= 100', 'color("purple", 0.5)'],
+     *             ['${Height} >= 50', 'color("red")'],
+     *             ['true', 'color("blue")']
+     *         ]
+     *     },
+     *     show : '${Height} > 0',
+     *     meta : {
+     *         description : '"Building id ${id} has height ${Height}."'
+     *     }
+     * });
+     */
+    function Cesium3DTileStyle(data) {
+        this._style = undefined;
+        this._ready = false;
+        this._readyPromise = when.defer();
+        this._color = undefined;
+        this._show = undefined;
+        this._pointSize = undefined;
+        this._meta = undefined;
+
+        this._colorShaderFunction = undefined;
+        this._showShaderFunction = undefined;
+        this._pointSizeShaderFunction = undefined;
+        this._colorShaderFunctionReady = false;
+        this._showShaderFunctionReady = false;
+        this._pointSizeShaderFunctionReady = false;
+
+        var style = this;
+        if (typeof data === 'string') {
+            RequestScheduler.request(data, loadJson).then(function(styleJson) {
+                setup(style, styleJson);
+                style._readyPromise.resolve(style);
+            }).otherwise(function(error) {
+                style._readyPromise.reject(error);
+            });
+        } else {
+            setup(style, data);
+            style._readyPromise.resolve(style);
+        }
+    }
+
+    function setup(that, styleJson) {
+        that._style = clone(styleJson, true);
+
+        styleJson = defaultValue(styleJson, defaultValue.EMPTY_OBJECT);
+
+        if (!defined(styleJson.color)) {
+            // If there is no color style do not create a shader function.
+            that._colorShaderFunctionReady = true;
+        }
+
+        if (!defined(styleJson.show)) {
+            // If there is no show style do not create a shader function.
+            that._showShaderFunctionReady = true;
+        }
+
+        if (!defined(styleJson.pointSize)) {
+            // If there is no point size style do not create a shader function.
+            that._pointSizeShaderFunctionReady = true;
+        }
+
+        var colorExpression = defaultValue(styleJson.color, DEFAULT_JSON_COLOR_EXPRESSION);
+        var showExpression = defaultValue(styleJson.show, DEFAULT_JSON_BOOLEAN_EXPRESSION);
+        var pointSizeExpression = defaultValue(styleJson.pointSize, DEFAULT_JSON_NUMBER_EXPRESSION);
+
+        var expressions = styleJson.expressions;
+
+        var color;
+        if (typeof colorExpression === 'string') {
```

Comment:
The default value is a string so I don't expect both to fail. If they do fail something is wrong with the json and that would be better caught by a validator.